docs/interceptors.md

```diff
@@ -0,0 +1,561 @@
+<!--
+---
+linkTitle: "Interceptors"
+weight: 5
+---
+-->
+# `Interceptors`
+
+- [Overview](#overview)
+- [Webhook `Interceptors`](#webhook-interceptors)
+- [GitHub `Interceptors`](#github-interceptors)
+- [GitLab `Interceptors`](#gitlab-interceptors)
+- [Bitbucket `Interceptors`](#bitbucket-interceptors)
+- [CEL `Interceptors`](#cel-interceptors)
+- [`ClusterInterceptors`](#clusterinterceptors)
+
+## Overview
```

Comment:
Interceptors are an integration point (sort of like custom tasks). So there are two parts to it:
1. For integrators, how do they add/implement new functionality. This is done in 2 ways (either by implementing a legacy webhook interceptor service or by implementing a new ClusterInterceptor one)
2. For trigger end users, how do they configure a interceptor within a Trigger. The legacy ways here are GitHub/GitLab/BitBucket/CEL/Webhook definitions. The new way is to refer to all interceptors using a combination of the `ref` and the `params` fields.

I think it makes sense to split the docs for the two kinds of users. We could have a section on how to implement an interceptor (i.e. ClusterInterceptor and the legacy webhook services). And we could have another section on how to use or invoke or configure an interceptor from within a Trigger.

The goal with the interceptor docs being split across two pages was to accomplish this -- though this was not perfect since the section in the eventlisteners file contained the part about implementing webhook interceptor services.

Reply:
I will rework the docs as per our chat; hold review until that's done. :)

Reply:
Reworked as per our chat.